pkgs/build-support/build-bazel-package/default.nix

```diff
@@ -50,13 +53,59 @@ let
   fArgs = removeAttrs args [ "buildAttrs" "fetchAttrs" "removeRulesCC" ];
   fBuildAttrs = fArgs // buildAttrs;
   fFetchAttrs = fArgs // removeAttrs fetchAttrs [ "sha256" ];
+  bazelCmd = { cmd, additionalFlags, targets }:
+    if targets == [ ] then "" else
+      ''
+          # Bazel sandboxes the execution of the tools it invokes, so even though we are
+          # calling the correct nix wrappers, the values of the environment variables
+          # the wrappers are expecting will not be set. So instead of relying on the
+          # wrappers picking them up, pass them in explicitly via `--copt`, `--linkopt`
+          # and related flags.
+          #
+          copts=()
+          host_copts=()
+          linkopts=()
+          host_linkopts=()
+          if [ -z "''${dontAddBazelOpts:-}" ]; then
+            for flag in $NIX_CFLAGS_COMPILE; do
+              copts+=( "--copt=$flag" )
+              host_copts+=( "--host_copt=$flag" )
+            done
+            for flag in $NIX_CXXSTDLIB_COMPILE; do
+              copts+=( "--copt=$flag" )
+              host_copts+=( "--host_copt=$flag" )
+            done
+            for flag in $NIX_LDFLAGS; do
+              linkopts+=( "--linkopt=-Wl,$flag" )
+              host_linkopts+=( "--host_linkopt=-Wl,$flag" )
+            done
+          fi
+
+          BAZEL_USE_CPP_ONLY_TOOLCHAIN=1 \
+          USER=homeless-shelter \
+          bazel \
+            --batch \
+            --output_base="$bazelOut" \
+            --output_user_root="$bazelUserRoot" \
+            ${cmd} \
+            --curses=no \
+            -j $NIX_BUILD_CORES \
+            "''${copts[@]}" \
+            "''${host_copts[@]}" \
+            "''${linkopts[@]}" \
+            "''${host_linkopts[@]}" \
+            $bazelFlags \
+            ${lib.strings.concatStringsSep " " additionalFlags} \
+            ${lib.strings.concatStringsSep " " targets}
+        '';
 
-in stdenv.mkDerivation (fBuildAttrs // {
-  inherit name bazelFlags bazelBuildFlags bazelFetchFlags bazelTarget;
+in
+stdenv.mkDerivation (fBuildAttrs // {
+  inherit doCheck name bazelFlags bazelBuildFlags bazelTestFlags bazelFetchFlags bazelTarget bazelTestTargets;
```

Comment:
This should be switch to pname+version at some point.

Reply:
It looks like there is the option to use pname+version as well (see [here](https://github.com/NixOS/nixpkgs/blob/6b1a33bede0e96f19b73e64de6dc4b5cbf400771/pkgs/build-support/build-bazel-package/default.nix#L12)).